### PR TITLE
Excluding remaining siloed content directories

### PR DESCRIPTION
**Pull request by flossypurse**
This PR excludes the remaining siloed content directories from the Docusaurus build.

This effectively removes the following slugs:
- `/go/*`
- `/java/*`
- `/clusters/*`
- `/concepts/*`

Internal links have been adjusted.

Redirects have been added for each group. These either redirect to the legacy docs for SDKs site - or to the other new guides in the docs.

### CODE REVIEW

File: docs-src/concepts/what-is-a-search-attribute.md [hunk 1]
```diff
@@ -131,4 +131,4 @@ Note:
 To actually have results from the use of a [List Filter](/concepts/what-is-a-list-filter), Search Attributes must be added to a Workflow Execution as metadata.
 How to do this entirely depends on the method by which you spawn the Workflow Execution:
 
-- [How to set Search Attributes as Workflow Execution metadata in Go](/go/startworkflowoptions-reference/#searchattributes)
+- [How to set Search Attributes as Workflow Execution metadata in Go](/app-dev-context/set-custom-search-attributes)
```
Comment (flossypurse) [inline docs-src/concepts/what-is-a-search-attribute.md:134]:
```suggestion
- [How to set Search Attributes as Workflow Execution metadata](/app-dev-context/set-custom-search-attributes)
```

[the remark this answers is not in the document]
Comment (flossypurse) [reply docs-src/concepts/what-is-a-search-attribute.md]: No, the text should be generic though.

File: vercel.json [hunk 2]
```diff
@@ -6,10 +6,10 @@
...
   "redirects": [
     {"source": "/docs/:path*", "destination": "/:path*"},
-    {
-      "source": "/typescript/:path*",
-      "destination": "https://legacy-documentation-sdks.temporal.io/typescript/:path*"
-    },
+    {"source": "/typescript/:path*", "destination": "https://legacy-documentation-sdks.temporal.io/typescript/:path*"},
+    {"source": "/go/:path*", "destination": "https://legacy-documentation-sdks.temporal.io/go/:path*"},
+    {"source": "/java/:path*", "destination": "https://legacy-documentation-sdks.temporal.io/java/:path*"},
+    {"source": "/concepts/what-is:path*", "destination": "concepts/landing"},
```
Comment (bergundy) [inline vercel.json:12]: Why redirect to the landing page here?

Comment (bergundy) [reply vercel.json:12]: Would be nice to keep the old URLs working.

**Review by bergundy — APPROVED** [review]

File: vercel.json (hunk 2 above)
Comment (flossypurse) [reply vercel.json:12]: @bergundy I will put in a redirect for each one before this merges.

**Review by djmagee — APPROVED** [review]

**Pull request merged**